### ISSUE TITLE
Could not load exchange rates errors

### ISSUE THREAD

**Issue by geneguido** [langerhans/dogecoin-wallet-new#46]
Hello, 

I get this error "Could not load exchange rates" a few months after installing the wallet. Can't see the value of my wallet anymore. This happened to me twice. Once with an earlier version and again with version 3.0.0. Not sure where this call is made in the code but if I did know I may be able to figure it out on my own. I have a few skills necessary just not familiar with this project.

**Comment by langerhans** [comment]
Calls are made here and it appears the Bter API is (currently?) not available anymore: https://github.com/langerhans/dogecoin-wallet-new/blob/master/wallet/src/de/schildbach/wallet/data/ExchangeRatesProvider.java

I guess that needs to be replaced with another API provider then...

**Comment by geneguido** [comment]
Thanks! I'll take a look into it.

**Comment by geneguido** [comment]
So far I've found a few replacement api's
all:
https://www.cryptonator.com/api/currencies
one:
https://api.cryptonator.com/api/ticker/doge-usd
benofits: it's free to use
drawbacks: It's got no specifics about what you can use it for. But the obvious reason for this service to exist is to promote cryptonator.com's online wallet. They may not like another wallet using them as source material. or they may never notice.

or

All:
https://api.coinmarketcap.com/v1/ticker/
Some:
https://api.coinmarketcap.com/v1/ticker/dogecoin/?convert=LTC
benofits: free to use, better api documentation, no obvious conflicts of interest.
drawbacks: They are a for profit company that i think makes money on ad revenue. So the give and take may be to list them as a provider in the app for this service somehow. They have very low api limits but they are still 5 times higher then we would need. So also not a big deal. 

As I have time I'll see what I can do. I am a bit slow coding but coinmarketcap.com looks to be the winner among api's i would even mention.

**Comment by langerhans** [comment]
Having used both I would prefer CMC. If you want to tackle this, then feel free and thanks in advance. I will probably update the wallet to upstream code soon-ish too so I could release a nice overall update :)

**Comment by geneguido** [comment]
Oh man, putting on the pressure j/k. 

Seriously though I've got a full schedule and work right up to Sunday. And no java IDE setup currently. It's been a while with java. So I am thinking minimum 1 week to get something done. But I'll keep you updated here.

**Comment by langerhans** [comment]
You'll need Android Studio for it. Also it's kinda hard to get it to compile in first place as you need to compile libdohj and bitcoinj from source to even get started. Let me know if you want me to walk you through this.

**Comment by PanderMusubi** [comment]
A bounty has been added at https://www.bountysource.com/issues/53147938-could-not-load-exchange-rates-errors

**Comment by langerhans** [comment]
Okay, while working on the rebase on the latest upstream changes I have also fixed this issue. The wallet will now use CMC for the conversion. If you already worked on it, sorry about that.

**Comment by geneguido** [comment]
Nah, if anything I should apologize for being incapable of fixing the issue quicker. I am just now getting free again to work on it lol. Thanks for the hard work!

**Comment by PanderMusubi** [comment]
@geneguido, any help on https://github.com/langerhans/dogecoin-wallet-new/issues/32 is also much appreciated.

**Comment by geneguido** [comment]
Thanks for the heads up. Unfortunately I am a bit useless right now. I do not have the skills required, and I also don't have enough time to spread around to get rolling on it. But I am going to keep an eye on this project and maybe in a month or two things will calm down after the specter and meltdown crap and I can actually help out a bit. Sorry guys.
